lib/puppet/functions/resources_deep_merge.rb

```diff
@@ -0,0 +1,87 @@
+# @summary Deeply merge a "defaults" hash into a "resources" hash like the ones expected by `create_resources()`.
+#
+# Deeply merge a "defaults" hash into a "resources" hash like the ones expected by `create_resources()`.
```

Comment:
@ekohl Updated to use `@summary`, but also repeat the summary line below so that it doesn't only appear in the TOC.